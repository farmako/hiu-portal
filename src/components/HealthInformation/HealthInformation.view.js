import React from 'react';
import * as PropTypes from 'prop-types';
import HealthInformationContent from './HealthInformationContent.view';

const HealthInformation = ({ consentReqId, healthInfo, selectedDate }) => {
  const healthInfoForSelectedDate = healthInfo ? healthInfo[selectedDate] : '';
  return healthInfoForSelectedDate ? (
    healthInfoForSelectedDate.map((hip) => (
      <HealthInformationContent
        key={hip.hipName}
        consentReqId={consentReqId}
        hipName={hip.hipName}
        data={hip.data}
      />
    ))
  ) : (
    <div />
  );
};

HealthInformation.propTypes = {
  healthInfo: PropTypes.shape({}),
  consentReqId: PropTypes.string,
  selectedDate: PropTypes.string,
};

HealthInformation.defaultProps = {
  healthInfo: null,
  consentReqId: '',
  selectedDate: '',
};

export default HealthInformation;
