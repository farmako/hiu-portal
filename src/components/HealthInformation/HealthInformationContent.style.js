import styled from 'styled-components';

const HealthInformationContentStyles = styled.div`
  .hip-health-info-container {
    font-family: "Roboto", sans-serif;
  }
  .header {
    text-transform: uppercase;
    letter-spacing: 1px;
    font-weight: 400;
    margin-bottom: 20px;
  }
`;

export default HealthInformationContentStyles;
