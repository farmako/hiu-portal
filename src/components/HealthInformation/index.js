import HealthInformation from './HealthInformation.view';

export default HealthInformation;
