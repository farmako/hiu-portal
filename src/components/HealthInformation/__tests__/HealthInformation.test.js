import React from 'react';
import { shallow } from 'enzyme';
import HealthInformation from '../HealthInformation.view';
import HealthInformationContent from '../HealthInformationContent.view';

describe('HealthInformation', () => {
  const mockHealthInfo = {
    '22/1/2020': [
      {
        hipId: '10000005',
        hipName: 'Max Health Care',
        data: [
          {
            resourceType: 'MedicationRequest',
            status: 'active',
            intent: 'order',
            medicationReference:
                            {
                              reference: 'Medication/c2b8f3fd-6da0-44ed-bc91-8bb05ff614d2',
                              display: 'IV antibiotics',
                            },
            subject:
                            {
                              display: 'Shriya',
                            },
            authoredOn: '2020-01-22',
            dosageInstruction: [
              {
                sequence: 1,
                text: '22/01/2020 to 07/02/2020',
                timing: {
                  code: {
                    text: '15 days',
                  },
                },
              },
            ],
          },
        ],
      },
    ],
  };
  const wrapper = shallow(
    <HealthInformation
      consentReqId="123"
      healthInfo={mockHealthInfo}
      selectedDate="22/1/2020"
    />,
  );

  it('should render properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render HealthInformationContent component with correct props', () => {
    expect(wrapper.find(HealthInformationContent).props().consentReqId).toEqual('123');
    expect(wrapper.find(HealthInformationContent).props().hipName).toEqual('Max Health Care');
    expect(wrapper.find(HealthInformationContent).props().data).toEqual(mockHealthInfo['22/1/2020'][0].data);
  });

  it('should render blank if data is undefined', () => {
    wrapper.setProps({ healthInfo: {} });
    expect(wrapper.find('div').length).toEqual(1);
  });
});
