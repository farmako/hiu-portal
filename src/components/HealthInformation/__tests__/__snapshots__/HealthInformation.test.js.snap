// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HealthInformation should render properly 1`] = `
<HealthInformationContent
  consentReqId="123"
  data={
    Array [
      Object {
        "authoredOn": "2020-01-22",
        "dosageInstruction": Array [
          Object {
            "sequence": 1,
            "text": "22/01/2020 to 07/02/2020",
            "timing": Object {
              "code": Object {
                "text": "15 days",
              },
            },
          },
        ],
        "intent": "order",
        "medicationReference": Object {
          "display": "IV antibiotics",
          "reference": "Medication/c2b8f3fd-6da0-44ed-bc91-8bb05ff614d2",
        },
        "resourceType": "MedicationRequest",
        "status": "active",
        "subject": Object {
          "display": "Shriya",
        },
      },
    ]
  }
  hipName="Max Health Care"
  key="Max Health Care"
/>
`;
